compiler/src/dotty/tools/dotc/core/ConstraintHandling.scala

```diff
@@ -102,7 +103,7 @@ trait ConstraintHandling[AbstractContext] {
           homogenizeArgs = Config.alignArgsInAnd
           try
             if (isUpper) oldBounds.derivedTypeBounds(lo, hi & bound)
-            else oldBounds.derivedTypeBounds(lo | bound, hi)
+            else oldBounds.derivedTypeBounds(typeLub(lo, bound), hi)
```

Comment:
Not sure this is enough in general, there's a few calls to `|` and `&` in https://github.com/lampepfl/dotty/blob/master/compiler/src/dotty/tools/dotc/core/Constraint.scala (even though the documentation of ConstraintHandling claims "Constraint code does not have access to a type-comparer. Anything regarding lubs and glbs has to be done elsewhere.")